lib/deep_cloning.rb

```diff
@@ -84,9 +93,10 @@ def walk?(cell)
           end
         else
           @opts[p.class_name] = {} unless @opts[p.class_name]
-          !cell.respond_to?("#{p.name}_id".to_sym) or cell.send("#{p.name}_id").nil? or @opts[p.class_name][cell.send("#{p.name}_id")] or p.class_name.in? @opts[:except] # replicated parent?
+          @must_ignore << p.class_name unless safe_child?(cell, p)
+          safe_child?(cell, p)
```

Comment:
We are calling safe child two times here!

Reply:
I’ll fox that as soon as I got home 🏡

Reply:
Fixed!